server/pulp/server/db/connection.py

```diff
@@ -143,6 +136,29 @@ def initialize(name=None, seeds=None, max_pool_size=None, replica_set=None, max_
         raise
 
 
+def _connect_to_one_of_seeds(connection_kwargs, seeds_list, db_name):
+    """
+    Helper function to iterate over a list of database seeds till a successful connection is made
+
+    :param connection_kwargs: arguments to pass to mongoengine connection
+    :param seeds_list: list of seeds to try connecting to
+    :return: Connection object if connection is made or None if no connection is made
+    """
+    for seed in seeds_list:
+        connection_kwargs.update({'host': seed})
```

Comment:
If you made this line:

```
connection_kwargs.update({'host': seed.strip()})
```

Then you could [remove this line](https://github.com/pulp/pulp/pull/2035/files#diff-cf72bb4351a6f36fccfff8dafaf8ff9dR49).